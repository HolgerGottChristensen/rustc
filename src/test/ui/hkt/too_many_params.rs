// HKT test for too many parameters
// check-fail

fn test<I<%J>>(input: I<u32, u32>) {} //~ ERROR this hkt parameter takes 1 generic argument but 2 generic arguments were supplied [E0107]

fn main() {
    test::<Option<%J>>(Some(42));
}
