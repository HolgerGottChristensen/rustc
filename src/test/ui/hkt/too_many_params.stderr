error[E0107]: this hkt parameter takes 1 generic argument but 2 generic arguments were supplied
  --> $DIR/too_many_params.rs:4:23
   |
LL | fn test<I<%J>>(input: I<u32, u32>) {}
   |                       ^      --- help: remove this generic argument
   |                       |
   |                       expected 1 generic argument
   |
note: hkt parameter defined here, with 1 generic parameter: `J`
  --> $DIR/too_many_params.rs:4:9
   |
LL | fn test<I<%J>>(input: I<u32, u32>) {}
   |         ^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0107`.
